Security/UserChecker.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the FOSUserBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\UserBundle\Security;
+
+use Symfony\Component\Security\Core\Exception\AccountExpiredException;
+use Symfony\Component\Security\Core\Exception\CredentialsExpiredException;
+use Symfony\Component\Security\Core\Exception\DisabledException;
+use Symfony\Component\Security\Core\Exception\LockedException;
+use Symfony\Component\Security\Core\User\UserChecker as BaseUserChecker;
+use Symfony\Component\Security\Core\User\UserInterface as BaseUserInterface;
+
+/**
+ * UserChecker checks the user account flags.
+ *
+ * @author Julian Finkler (Devtronic) <julian@developer-heaven.de>
+ */
+class UserChecker extends BaseUserChecker
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function checkPreAuth(BaseUserInterface $user)
+    {
+        if (!$user->isAccountNonLocked()) {
```

Comment:
IMO, the FOSUserBundle checker should check only the methods that belong to FOSUserBundle (account locking is not part of FOSUserBundle)

Reply:
The issue is that some people might be using these methods in their app. Removing it would break their application.